charm-slurmd/src/charm.py

```diff
@@ -529,6 +531,11 @@ def singularity_install(self, event):
                 logger.error(f"## Missing singularity resource - {e}")
                 event.fail(message=f'Error installing Singularity: {e.output}')
 
+    def mpi_install(self, event):
+        """Install MPI (mpich)."""
+        self._slurm_manager.mpi.install()
+        event.set_results({'installation': 'Successfull.'})
```

Comment:
If there's an error in the installation, the action will still return a success message?

Reply:
Good observation. I followed the Nvidia example for this: https://github.com/omnivector-solutions/slurm-charms/blob/master/charm-slurmd/src/charm.py#L508

Reply:
Fixed on the latest commit